crates/aranya-policy-ifgen-build/src/imp.rs

```diff
@@ -15,8 +15,14 @@ pub fn generate_code(policy: &Policy) -> String {
         .map(|s| {
             let doc = format!(" {} policy struct.", s.identifier);
             let name = mk_ident(&s.identifier);
-            let names = s.fields.iter().map(|f| mk_ident(&f.identifier));
-            let types = s.fields.iter().map(|f| vtype_to_rtype(&f.field_type));
+            let names = s.items.iter().map(|i| match i {
+                StructItem::Field(f) => mk_ident(&f.identifier),
+                StructItem::StructRef(_) => todo!(),
+            });
+            let types = s.items.iter().map(|i| match i {
+                StructItem::Field(f) => vtype_to_rtype(&f.field_type),
+                StructItem::StructRef(_) => todo!(),
```

Comment:
@jdygert-spok I'm not sure what to do here... This runs before the compiler has had a chance to resolve the struct references to fields, and we don't have access to the Module, so we can't look up the struct ourselves.

Reply:
Actually, we can look up the struct in the policy (`policy.structs`). So, would we have to duplicate the compiler's field insertion logic? Then remove the struct reference from the policy struct so the compiler doesn't try to insert the fields a second time?